quickjs.c

```diff
@@ -209,7 +209,7 @@ typedef enum {
     JS_GC_PHASE_REMOVE_CYCLES,
 } JSGCPhaseEnum;
 
-typedef enum OPCodeEnum OPCodeEnum;
+__extension__ typedef enum OPCodeEnum OPCodeEnum;
```

Comment:
The forward declaration for `OPCodeEnum` does not seem required as the name is not used before the actual definition around line 980.  Try removing  line 212 and patching the definition as a `typedef`.